silq/instrument_interfaces/keysight/E8267D_interface.py

```diff
@@ -378,12 +415,15 @@ def get_additional_pulses(self, interface: InstrumentInterface):
                     "FM_mode = 'ramp' and IQ_modulation = 'on'"
 
                 frequency = self.pulse.frequency + self.pulse.frequency_sideband
-                frequency_difference = frequency - interface.frequency()
-                amplitude_FM = frequency_difference / interface.frequency_deviation()
+                frequency_difference = frequency - settings['frequency']
+                amplitude_FM = frequency_difference / settings['frequency_deviation']
                 frequency_IQ = self.pulse.frequency_sideband
             else:  # interface.FM_mode() == 'IQ'
                 amplitude_FM = None
-                frequency_IQ = self.pulse.frequency - interface.frequency()
+                frequency_IQ = self.pulse.frequency - settings['frequency']
+
+        attenuation = self.pulse.power - settings['power']
+        amplitude = 10.0 ** (attenuation / 20)
```

Comment:
I understand that you subtract maximum power, but would still add a check for amplitude limitation to 1V.